### PR TITLE
Add EDGEDB_SERVER_DOCKER_LOG_LEVEL to entrypoint

### PR DESCRIPTION
**Pull request by fmoor**
working towards https://github.com/edgedb/edgedb-cli/issues/510

### CODE REVIEW

File: docker-entrypoint-funcs.sh [hunk 1]
```diff
@@ -1,10 +1,16 @@
 #!/usr/bin/env bash
 
+declare -A log_levels=([trace]=0, [info]=1, [warning]=2)
+_script_log_level="info"
```
Comment (elprans) [inline docker-entrypoint-funcs.sh:4]:
```suggestion
declare -A _edbdocker_log_levels=([trace]=0, [info]=1, [warning]=2)
_edbdocker_log_level="info"
```

This file is intended to be `source`-d into other scripts, so all variables and functions must be namespaced carefully.

File: README.md [hunk 2]
```diff
@@ -164,6 +164,11 @@ user and database creation, and other initial setup. If neither the
 will look for the presence of `/edgedb-bootstrap.edgeql` in the container
 (which can be placed in a derived image).
 
+#### `EDGEDB_SERVER_DOCKER_LOG_LEVEL`
+
+Determines the log verbosity level in the entrypoint script. Valid levels are
+`trace`, `info`, and `warning`
```
Comment (elprans) [inline README.md:170]:
```suggestion
`trace`, `info`, and `warning`.  The default is `info`.
```

File: docker-entrypoint-funcs.sh [hunk 3]
```diff
@@ -1,10 +1,16 @@
 #!/usr/bin/env bash
 
+declare -A _edbdocker_log_levels=([trace]=0, [debug]=1, [info]=2, [warning]=3)
+_edbdocker_log_level="info"
 
 edbdocker_setup_shell() {
   set -Eeo pipefail
   shopt -s dotglob inherit_errexit nullglob compat"${BASH_COMPAT=42}"
-  if [ -n "$EDGEDB_SERVER_DOCKER_TRACE" ]; then
+  if [[ -n ${EDGEDB_SERVER_DOCKER_LOG_LEVEL+x} ]]; then
```
Comment (elprans) [inline docker-entrypoint-funcs.sh:9]: Wouldn't this result in `_edbdocker_log_level` being set to an empty string if I pass `env EDGEDB_SERVER_DOCKER_LOG_LEVEL=""`?

Comment (fmoor) [reply docker-entrypoint-funcs.sh:9]: I added a check for invalid log level values.

**Review by elprans — APPROVED** [review]

**Pull request merged**